lib/matplotlib/tests/test_offsetbox.py

```diff
@@ -470,3 +470,32 @@ def test_draggable_in_subfigure():
     bbox = ann.get_window_extent()
     MouseEvent("button_press_event", fig.canvas, bbox.x1+2, bbox.y1+2)._process()
     assert not ann._draggable.got_artist
+
+
+
+def test_anchored_offsetbox_tuple_and_float_borderpad():
+    """
+    Test AnchoredOffsetbox correctly handles both float and tuple for borderpad.
+    """
+    from matplotlib.offsetbox import AnchoredText
+
+    fig, ax = plt.subplots()
+
+    # Case 1: Test the old float behavior (exercises the 'except' block)
+    text_float = AnchoredText("float", loc='lower left', borderpad=5)
+    ax.add_artist(text_float)
+
+    # Case 2: Test the new tuple behavior (exercises the 'try' block)
+    text_tuple = AnchoredText("tuple", loc='lower left', borderpad=(10, 10))
+    ax.add_artist(text_tuple)
```

Comment:
```suggestion
    # Case 1: Test the float behavior
    text_float = AnchoredText("float", loc='lower left', borderpad=5)
    ax.add_artist(text_float)

    # Case 2: Test the tuple behavior
    text_tuple = AnchoredText("tuple", loc='lower left', borderpad=(10, 10))
    ax.add_artist(text_tuple)
```
"Old" and "new" only make sense in the context of this PR, and not to someone reading this in the future.  I also think it's better not to comment about how it's currently implemented: if someone refactors the code at some point, they might just confirm the tests still pass without reading and updating the test comments.